src/s1reader/s1_burst_slc.py

```diff
@@ -106,6 +106,80 @@ class represents a polynomial function of range
     poly = isce3.core.Poly2d(coeffs, xmin, ymin, xnorm, ynorm)
     return poly
 
+def _evaluate_polynomial_array(arr_polynomial, grid_tau, vec_tau0):
+    '''
+    Evaluate the polynomials on the correction grid.
+    To be used for calculating azimuth FM mismatch rate.
+
+    Parameters:
+    -----------
+    arr_polynomial: np.ndarray
+        coeffieients interpolated to azimuth times in each line
+        in correction grid
+    grid_tau: np.ndarray
+        2d numpy array filled with range time in the correction grid
+    vec_tau0: np.ndarray
+        range start time for each line in the correction grid
+
+    Return:
+    -------
+    eval_out: np.ndarray
+        Evaluated values on the correction grid
+    '''
+
+    ncol = grid_tau.shape[1]
+    term_tau = grid_tau - vec_tau0 * np.ones(ncol)[np.newaxis, ...]
+
+    eval_out = (arr_polynomial[:,0][..., np.newaxis]
+                *np.ones(ncol)[np.newaxis, ...]
+                + (arr_polynomial[:,1][..., np.newaxis]
+                    * np.ones(ncol)[np.newaxis, ...]) * term_tau
+                + (arr_polynomial[:,2][..., np.newaxis]
+                    * np.ones(ncol)[np.newaxis, ...]) * term_tau**2)
+
+    return eval_out
+
+def _llh_to_ecef(lat, lon, hgt, ellipsoid, unit_degree=True):
+    '''
+    Calculate cartesian coordinates in ECEF from
+    latitude, longitude, and altitude
+
+    Parameters:
+    -----------
+    lat: np.ndarray
+        latitude as numpy array
+    lon: np.ndarray
+        longitude as numpy array
+    hgt: np.ndarray
+        height as numpy array
+    ellipsoid: isce3.core.Ellipsoid
+        Definition of Ellipsoid
+    unit_degree: bool
```

Comment:
```suggestion
    in_degree: bool
```
(so that the check reads `if in_degrees:` rather than `if unit_degrees`)